internal/autoupdate/queue.go

```diff
@@ -473,6 +473,33 @@ func (q *queue) updatePR(ctx context.Context, pr *PullRequest) {
 	ctx, cancelFunc := context.WithTimeout(ctx, retryTimeout)
 	defer cancelFunc()
 
+	isApproved, err := q.pullRequestIsApproved(ctx, pr)
+	if err != nil {
+		logger.Error(
+			"checking if pr is approved failed",
+			logfields.Event("approval_state_check_failed"),
+			zap.Error(err),
+		)
+		return
+	}
+
+	if !isApproved {
+		if err := q.Suspend(pr.Number); err != nil {
+			logger.Error(
+				"suspending PR because it is not approved , failed",
+				logfields.Event("suspending_pr_updates_failed"),
+				zap.Error(err),
+			)
+			return
+		}
+		logger.Info(
+			"updates suspended, pr is not approved",
+			logFieldReason("pr_not_approved"),
+			logEventUpdatesSuspended,
+			zap.Error(err),
+		)
+	}
```

Comment:
Shouldn't it return here even if `Suspend()` is not an error?

Reply:
definitely, nice catch, thanks!

Reply:
fixed